compiler/rustc_mir_build/src/thir/pattern/check_match.rs

```diff
@@ -405,7 +405,8 @@ impl<'p, 'tcx> MatchVisitor<'p, 'tcx> {
         scrut_ty: Ty<'tcx>,
     ) -> Result<UsefulnessReport<'p, 'tcx>, ErrorGuaranteed> {
         let pattern_complexity_limit =
-            get_limit_size(cx.tcx.hir().krate_attrs(), cx.tcx.sess, sym::pattern_complexity);
+            get_limit_size(cx.tcx.hir().krate_attrs(), cx.tcx.sess, sym::pattern_complexity)
+            .or(Some(10_000_000)); // Default value to emit the warning for "too complex" match.
```

Comment:
Could you quickly mention how the value was chosen in a comment?

Reply:
Sure, done!